src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -0,0 +1,165 @@
+/* @internal */
+namespace ts.refactor.generateGetAccessorAndSetAccessor {
+    const actionName = "Generate 'get' and 'set' accessors";
+    const actionDescription = Diagnostics.Generate_get_and_set_accessors.message;
+    registerRefactor(actionName, { getEditsForAction, getAvailableActions });
+
+    interface Info {
+        originalName: string;
+        fieldName: string;
+        accessorName: string;
+        accessorType: TypeNode;
+        propertyDeclaration: PropertyDeclaration;
+        needUpdateName: boolean;
+        hasModifiers: boolean;
+        needUpdateModifiers: boolean;
+    }
+
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        return [
+            {
+                name: actionName,
+                description: actionDescription,
+                actions: [
+                    {
+                        name: actionName,
+                        description: actionDescription
+                    }
+                ]
+            }
+        ];
+    }
+
+    function getEditsForAction(context: RefactorContext, _actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+        const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+
+        const { fieldName, accessorName, accessorType, propertyDeclaration, needUpdateName, hasModifiers, needUpdateModifiers } = fieldInfo;
+        const accessorModifiers = hasModifiers ? (

```

Comment:
`hasModifiers` currently is true if *any* member on the class has modifiers. This could just preserve whatever modifier was there before on the particular member without looking at other members. I.e., `public x: number` becomes `public get x() { ... } public set x() { ... }` and `x: number` becomes `get x() { ... } set x() { ... }`, making `hasModifiers` unnecessary.